src/feeds/CurveLPPessimisticFeed.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import {ICurvePool} from "src/interfaces/ICurvePool.sol";
+import {IChainlinkBasePriceFeed} from "src/interfaces/IChainlinkFeed.sol";
+
+contract CurveLPPessimisticFeed {
+    ICurvePool public immutable curvePool;
+
+    IChainlinkBasePriceFeed public immutable coin1Feed;
+
+    IChainlinkBasePriceFeed public immutable coin2Feed;
+
+    constructor(address _curvePool, address _coin1Feed, address _coin2Feed) {
+        curvePool = ICurvePool(_curvePool);
+        coin1Feed = IChainlinkBasePriceFeed(_coin1Feed);
+        coin2Feed = IChainlinkBasePriceFeed(_coin2Feed);
+    }
+
+    /**
+     * @return roundId The round ID of the Chainlink price feed
+     * @return minLpUsdprice The latest LP token price in USD
+     * @return startedAt The timestamp when the latest round of Chainlink price feed started
+     * @return updatedAt The timestamp when the latest round of Chainlink price feed was updated
+     * @return answeredInRound The round ID in which the answer was computed
+     */
+    function latestRoundData()
+        public
+        view
+        returns (uint80, int256, uint256, uint256, uint80)
+    {
+        (
+            uint80 roundId,
+            int256 usdPriceCoin1,
+            uint startedAt,
+            uint updatedAt,
+            uint80 answeredInRound
+        ) = coin1Feed.latestRoundData();
+
+        (
+            uint80 roundIdCoin2,
+            int256 usdPriceCoin2,
+            uint startedAtCoin2,
+            uint updatedAtCoin2,
+            uint80 answeredInRoundCoin2
+        ) = coin2Feed.latestRoundData();
+
+        int256 minLpUsdPrice;
+
+        // If coin1 price is lower than coin2 price, use coin1 price
+        if ((usdPriceCoin1 < usdPriceCoin2 && updatedAt > 0)) {
```

Comment:
Are you sure that, in theory, pool shares (LPs) always get minted at a rate that is <= 1:1 ?

i.e. imagine if some pool type mints 100 shares to someone who deposits 1USDC:1DAI ($2 value). the feed would show ~$100 value.

i'm not aware of Curve pools where this is the case, but worth checking since there's so many types now. Tricrypto, V2, volatile pairs, etc.

Reply:
sounds good. as far as i know, you are correct with curve mechanics. what troubles me about the assumption is seeing velodrome/aerodrome pools which issue LPs at distinctly different prices than 1:1. but should be good as long as things are checked case by case.

refer to this pool which has $800k TVL, and each 1 LP is priced at $2,000,156,024,660.25
https://basescan.org/token/0x27a8Afa3Bd49406e48a074350fB7b2020c43B2bD